src/workerd/api/util.c++

```diff
@@ -225,10 +225,10 @@ kj::String redactUrl(kj::StringPtr url) {
     // character.
 
     if (isUpper || isLower || isDigit || isSep) {
-      if (isHexDigit) {
+      if (isHexDigit(c)) {
         hexDigitCount++;
       }
-      if (!isHexDigit && !isSep) {
+      if (!isHexDigit(c) && !isSep) {
```

Comment:
IMHO: These 2 if statements should be merged since they're making 2 lookups, and previously they didn't.